[13.0][MIG] hr_contract

Yasaie opened the pull request:
Migration of `hr_contract`
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Yasaie committed:
[MIG] hr_contract

MiquelRForgeFlow commented on addons/hr_contract/migrations/13.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,35 @@
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_contract hc
+        SET company_id = c.id
+        FROM (SELECT id FROM res_company ORDER BY id LIMIT 1) AS c
+        WHERE hc.company_id IS NULL
```
```suggestion
        SET company_id = ru.company_id
        FROM res_users ru
        WHERE ru.id = hc.create_uid AND hc.company_id IS NULL
```

MiquelRForgeFlow commented on addons/hr_contract/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,43 @@
+hr_contract  / hr.contract              / kanban_state (selection)      : NEW selection_keys: ['blocked', 'done', 'normal'], hasdefault
+hr_contract  / hr.contract              / reported_to_secretariat (boolean): DEL
+hr_contract  / hr.contract              / state (selection)             : selection_keys is now '['cancel', 'close', 'draft', 'open']' ('['cancel', 'close', 'draft', 'open', 'pending']')
+# NOTHING TO DO
```
map state = 'pending' to state='open'

MiquelRForgeFlow commented on addons/hr_contract/migrations/13.0.1.0/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,43 @@
+hr_contract  / hr.contract              / company_id (many2one)         : now required, req_default: function
+# DONE: post-migration: filled the field when empty
+hr_contract  / hr.contract              / hr_responsible_id (many2one)  : NEW relation: res.users
+hr_contract  / hr.contract              / kanban_state (selection)      : NEW selection_keys: ['blocked', 'done', 'normal'], hasdefault
```
map old kanban_state = 'pending' to state='blocked'

MiquelRForgeFlow commented on addons/hr_contract/migrations/13.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,35 @@
+    contract = env['hr.contract']
+    employees = env['hr.employee'].search([])
+    for employee in employees:
+        employee.contract_id = contract.search([('employee_id', '=', employee.id)], order='date_start desc', limit=1)
```
```suggestion
        employee.contract_id = contract.search([('employee_id', '=', employee.id), ('company_id', '=', employee.company_id.id)], order='date_start desc', limit=1)
```
cannot be done using sql queries? if not, then do read_group or alike, because a `search` in a `for` is not good.

Yasaie commented:
@MiquelRForgeFlow Thanks for reviews. changes made

MiquelRForgeFlow reviewed (approved):
LGTM

pedrobaeza committed:
[MIG+FIX] hr_contract: Finish migration scripts

- Don't delete obsolet model records
- Refine contract company_id query
- Refine contract_id assignation in employees

pedrobaeza commented:
Added some refinements, and green, so merging.
